src/Tokenizer/Tokens.php

```diff
@@ -1195,7 +1195,7 @@ public function clearTokenAndMergeSurroundingWhitespace($index)
      */
     private static function calculateCodeHash($code)
```

Comment:
What about dropping this private method altogether?

Reply:
let us hotfix issue and decouple Tokens with a plan, not ad hoc

Reply:
Not sure I understand your comment. The changes here make `Tokens::calculateCodeHash` useless as it's just a proxy to another static method, so we could call it directly.

Reply:
Ditched it and did some other minors.
I wonder though, could we somehow prevent that the `linters` need to know that they should clear the cache before checking `code`?  hmm...

Reply:
I wanted to have single point of integration between Tokens and hasher.

yes, we can, we could extract whole caching out of Tokenizer and make caching decorator.

totally out of scope, even if nice to have

Reply:
agreed on all points :+1:  so let us put it on the roadmap :)
hope you are with the intermediate PR to deal with the current issue, we can deal with the rest later I think?

Reply:
that's why I suggested few comments above to not try to refactor the code ...

Reply:
check, I'll revert, hold my beer

Reply:
done